hack/save-logs-files.sh

```diff
@@ -0,0 +1,236 @@
+mkdir $dirname_logs_by_container
+
+is_found_fybrik_application_yaml=0
+
+# update fybrik application variables and check flags usage
+if [[ ! -z $fybrik_application_uuid ]]; then
+    # uuid flag is used
+    if [[ ! -z $fybrik_application_name ]]; then
+        # application flag is also used
+        echo "error: $APP_CONFLICT_MESSAGE"
+        print_usage
+        exit 1
+    fi
+else
+    if [[ ! -z $fybrik_application_name ]]; then
+        # only application flag is used
+        kubectl get fybrikapplication $fybrik_application_name -n $fybrik_application_ns -o=yaml &> $dirname_logs/fybrik_application_$fybrik_application_ns--$fybrik_application_name.yaml
+        is_found_fybrik_application_yaml=1
+        fybrik_application_uuid=$(kubectl get fybrikapplication $fybrik_application_name -n $fybrik_application_ns -o=jsonpath='{.metadata.uid}')
+    fi
+fi
+
+# iterate over the relevant namespaces
+for ns in ${namespaces_list[@]}; do
+    # save the logs of each container separately
+    for pod in $(kubectl get pods -n $ns -o=jsonpath='{.items[*].metadata.name}'); do
+        containers=$(kubectl get pods $pod -n $ns -o jsonpath='{.spec.containers[*].name}')
+        for container_name in $containers; do
+            kubectl logs $pod -n $ns -c $container_name --since=$duration --timestamps &> $dirname_logs_by_container/$ns--$pod--$container_name.txt
+        done
+    done
+    # save the non-certificates configmaps
+    for cm in $(kubectl get cm -n $ns -o=jsonpath='{.items[*].metadata.name}'); do
+        if [[ ! "$cm" == *.crt ]]; then
+            cm_data=$(kubectl get cm $cm -n $ns -o=jsonpath='{.data}')s
+            prefix=$ns--$cm
+            echo -e "$prefix:\n$cm_data\n" >> $dirname_logs/configmaps.txt
+        fi
+    done
+    # save output of 'get pods' command
+    get_pods_output=$(kubectl get pods -n $ns 2>&1)
+    echo -e "$ns:\n$get_pods_output\n" >> $dirname_logs/get_pods_output.txt
+    # save yamls of deployed modules
+    for fybrik_module in $(kubectl get fybrikmodules -n $ns -o=jsonpath='{.items[*].metadata.name}'); do
+        kubectl get fybrikmodule $fybrik_module -n $ns -o yaml &> $dirname_logs/fybrik_module_$ns--$fybrik_module.yaml
+    done
+    # if given fybrik application uuid, save its yaml and create another file which only contains logs with the uuid
+    if [[ ! -z fybrik_application_uuid ]] && [[ is_found_fybrik_application_yaml -eq 0 ]]; then
+        for fybrik_application in $(kubectl get fybrikapplications -n $ns -o=jsonpath='{.items[*].metadata.name}'); do
+            if [[ $(kubectl get fybrikapplication $fybrik_application -n $ns -o=jsonpath='{.metadata.uid}') == $fybrik_application_uuid ]]; then
+                kubectl get fybrikapplication $fybrik_application -n $ns -o=yaml &> $dirname_logs/fybrik_application_$ns--$fybrik_application.yaml
+                is_found_fybrik_application_yaml=1
+            fi
+        done
+    fi
+done
```

Comment:
when I ran the script without any arguments I got the following message:
`./save-logs-files.sh: line 186: namespaces_list[@]: unbound variable`

Reply:
This doesn't happen on linux, this is probably because of an old version of `bash` in macOS: https://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u
For now it was decided to not support mac.